src/index.ts

```diff
@@ -0,0 +1,612 @@
+    });
+
+    api.onBeforeBuild(async () => {
+      const { $ } = await import('execa');
+       $`npx --yes react-router typegen`;
+    });
+
+    const jiti = createJiti(process.cwd());
+
+    // Read the react-router.config.ts file first
+    const {
+      appDirectory = 'app',
+      basename = '/',
+      buildDirectory = 'build',
+      ssr = true,
+    } = await jiti
+      .import<Config>('./react-router.config.ts', {
+        default: true,
+      })
+      .catch(() => {
+        console.error(
+          'No react-router.config.ts found, using default configuration.',
+        );
+        return {} as Config;
+      });
+
+    const routesPath = resolve(appDirectory, 'routes.ts');
+
+    // Then read the routes
+    const routeConfig = await jiti
+      .import<RouteConfigEntry[]>(routesPath, {
+        default: true,
+      })
+      .catch((error) => {
+        console.error('Failed to load routes.ts:', error);
+        console.error('No routes.ts found in app directory.');
+        return [] as RouteConfigEntry[];
+      });
+
+    const jsExtensions = ['.tsx', '.ts', '.jsx', '.js', '.mjs'];
+
+    const findEntryFile = (basePath: string) => {
+      for (const ext of jsExtensions) {
+        const filePath = `${basePath}${ext}`;
+        if (existsSync(filePath)) {
+          return filePath;
+        }
+      }
+      return `${basePath}.tsx`; // Default to .tsx if no file exists
+    };
+
+    const entryClientPath = findEntryFile(
+      resolve(appDirectory, 'entry.client'),
+    );
+    const entryServerPath = findEntryFile(
+      resolve(appDirectory, 'entry.server'),
+    );
+
+    // Check for server app file
+    const serverAppPath = findEntryFile(
+      resolve(appDirectory, '../server/app'),
+    );
+    const hasServerApp = existsSync(serverAppPath);
+
+    // Add fallback logic for entry files
+    const templateDir = resolve(__dirname, 'templates');
+    const templateClientPath = resolve(templateDir, 'entry.client.js');
+    const templateServerPath = resolve(templateDir, 'entry.server.js');
+
+    // Use template files if user files don't exist
+    const finalEntryClientPath = existsSync(entryClientPath)
+      ? entryClientPath
+      : templateClientPath;
+    const finalEntryServerPath = existsSync(entryServerPath)
+      ? entryServerPath
+      : templateServerPath;
+
+    const rootRouteFile = relative(
+      appDirectory,
+      resolve(appDirectory, 'root.tsx'),
+    );
+
+    const routes = {
+      root: { path: '', id: 'root', file: rootRouteFile },
+      ...configRoutesToRouteManifest(appDirectory, routeConfig),
+    };
+
+    const outputClientPath = resolve(buildDirectory, 'client');
+    const assetsBuildDirectory = relative(process.cwd(), outputClientPath);
+
+    let clientStats: Rspack.StatsCompilation | undefined;
+    api.onAfterEnvironmentCompile(({ stats, environment }) => {
+      if (environment.name === 'web') {
+        clientStats = stats?.toJson();
```

Comment:
Getting all the stats in Rspack is quite slow, if we can specify options it will be faster.

see https://rspack.dev/config/stats#statsall